Core/include/Acts/Detector/DetectorVolume.hpp

```diff
@@ -417,11 +417,19 @@ class DetectorVolumeFactory {
       const std::vector<std::shared_ptr<Surface>>& surfaces,
       const std::vector<std::shared_ptr<DetectorVolume>>& volumes,
       DetectorVolumeUpdater detectorVolumeUpdater,
-      SurfaceCandidatesUpdater surfaceCandidateUpdater) {
+      SurfaceCandidatesUpdater surfaceCandidateUpdater, int nseg = 1000) {
```

Comment:
This will make the test extremely slow, I understand the intention, of course ...

Reply:
Here's an idea:
- we could make the containment check optional, e.g. if `nseg = -1` then now checking is done, so in a production workflow (after you have checked your geometry) you can skip the test in each job ?

What do you think?